src/components/vouchers/VoucherPrint.js

```diff
@@ -0,0 +1,132 @@
+import VoucherListTable from "./VoucherListTable";
+
+export default class VoucherPrint extends Component {
+  translateFilter() {
+    const firstVoucher = this.props.vouchers[0];
+
+    const {
+      taxi_id,
+      company_id,
+      employee_id,
+      created_start_at,
+      created_end_at
+    } = this.props.filterValues;
+
+    let filterParams = {};
+
+    if (taxi_id !== undefined) {
+      filterParams = { ...filterParams, taxi: firstVoucher.taxi.smtt };
+    }
+
+    if (company_id !== undefined) {
+      filterParams = { ...filterParams, company: firstVoucher.company.name };
+    }
+
+    if (employee_id !== undefined) {
+      filterParams = { ...filterParams, employee: firstVoucher.employee.name };
+    }
+
+    if (employee_id !== undefined) {
+      filterParams = { ...filterParams, employee: firstVoucher.employee.name };
+    }
+
+    if (created_start_at !== undefined) {
+      filterParams = { ...filterParams, created_start_at: created_start_at };
+    }
+
+    if (created_end_at !== undefined) {
+      filterParams = { ...filterParams, created_end_at: created_end_at };
+    }
+
+    return filterParams;
+  }
+
+  renderFilterHeader(filterParams) {
+    return (
+      <div className="ui medium labels">
+        {filterParams.created_start_at && (
+          <div className="ui label">
+            <i className="calendar alternate icon"></i>
+            Data Inicio:
+            <span className="detail">{filterParams.created_start_at}</span>
+          </div>
+        )}
+        {filterParams.created_end_at && (
+          <div className="ui label">
+            <i className="calendar alternate icon"></i>
+            Data Fim:
+            <span className="detail">{filterParams.created_end_at}</span>
+          </div>
+        )}
+        {filterParams.taxi && (
+          <div className="ui label">
+            <i className="taxi icon"></i>
+            Taxi:
+            <span className="detail">{filterParams.taxi}</span>
+          </div>
+        )}
+        {filterParams.company && (
+          <div className="ui label">
+            <i className="building icon"></i>
+            Empresa:
+            <span className="detail">{filterParams.company}</span>
+          </div>
+        )}
+        {filterParams.employee && (
+          <div className="ui label">
+            <i className="users icon"></i>
+            Funcionário:
+            <span className="detail">{filterParams.employee}</span>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  renderTotalValue() {
+    const totalValues = this.props.vouchers.reduce(
+      (totalValues, voucher) => totalValues + parseInt(voucher.value, 10),
+      0
+    );
+
+    return (
+      <div className="ui label big">
+        <i className="dollar icon"></i>
+        Valor Total:
+        <span className="detail">{formatCurrency(totalValues)}</span>
+      </div>
+    );
+  }
+
+  render() {
+    if (this.props.vouchers && this.props.vouchers.length > 0) {
```

Comment:
Não faz mais sentido essa condição estar no componente que chama esse componente, já que se for falsa não renderiza nada?

Reply:
rpz, acho q responsabilidade de saber oq vai renderizar é do proprio componente, q acha @thiagogsr ?

Reply:
isso @eliasousa, esse componente precisa viver e funcionar independente de outro, nada impede de colocar no componente pai **também**, caso queira